### ISSUE TITLE
Updating nested objects and then deleting the object gives error: TypeError: Cannot read property 'remove' of undefined

### ISSUE THREAD

**Issue by anuragvohraec** [Alex-Werner/SBTree#20]
### Expected Behavior

If a nested object is updated, it should be allowed to be deleted too.

### Current Behavior

If we update the nested object and then try and delete it, it will throw error "TypeError: Cannot read property 'remove' of undefined".

### Context
I am creating a cache based on SBTree and have to update nested docs in those cache.
So I found out about this bug.

### Steps to Reproduce & logs
```js
const {expect} = require('chai');
const {mocha, done} = require('mocha');
const SBTree = require("../../src/types/SBTree/SBTree");


describe("Test Update document and then delete", async (done)=>{
    const sbtree = new SBTree({order: 3});

    //Inserting a doc
    const t1 = await sbtree.insertDocuments({
        id: 'KqXIhr',
        key: 'KqXIhr',
        value: {
          _id: 'KqXIhr',
          _rev: '1-85e8ff864ae6d157217ad2e4d4117f5e',
        },
        ts: 1589885726633,
        _id: '5ec3bb1e299eb9b64166ebbb'
      });

      //finding the doc
    const [t2] = await sbtree.findDocuments({key: 'KqXIhr'});
    console.log("Found: ",t2);
    
    //updating the doc and replacing it 
    t2.value.count=1;
    let time_now = new Date().getTime();
    const t3 = await sbtree.replaceDocuments({...t2 , ts: new Date().getTime()});
    console.log(t3);
    
    //trying to delete the doc
    const t4 = await sbtree.deleteDocuments({ts: {$lte: time_now+1000}});
    console.log("deleted: ", t4)
})
```

Error stack trace I get is this : [line number will not be correct, as I have instered few console.logs]
```
(node:13503) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'remove' of undefined
    at removeNestedProp (/home/anurag/Documents/projects/project_dwarpal/used_libs/SBTree/src/types/SBTree/ops/remove.js:18:23)
    at SBTree.remove (/home/anurag/Documents/projects/project_dwarpal/used_libs/SBTree/src/types/SBTree/ops/remove.js:55:21)
    at async SBTree.deleteDocuments (/home/anurag/Documents/projects/project_dwarpal/used_libs/SBTree/src/types/SBTree/methods/deleteDocuments.js:13:11)
    at async Suite.<anonymous> (/home/anurag/Documents/projects/project_dwarpal/used_libs/SBTree/test/bugfixes/test1.js:32:16)
```

**Comment by anuragvohraec** [comment]
This is probably occuring as no field trees are added for feilds added to a document at runtime.
In above example we have added, **count** field in the object stored in doc. No runtme field tree is added for it.

Is it intended design and will it cause major code change  [In that case I will redesign my cache]?

**Comment by Alex-Werner** [comment]
This should be fixed now (in v3.0.1). 
And this exact scenario is now part of the functional tests, more complex tests need to be done (incursion of nested doc in nested doc).
